### PR TITLE
Check response status code in classes that extend RssParser

### PR DESCRIPTION
**Pull request by mynameisbogdan**
#### Database Migration
NO

### CODE REVIEW

**Comment by Qstick** [comment]
And info on what we are attempting to solve here? Torznab spec indicates we should always get a 200 back, and the exception, 429s are handled in HttpClient.

**Comment by Qstick** [comment]
Ok, so issues in upstream app/site that cause 500 errors to be returned. We should probably add to NewsnabRssParser as well in that case, as a 500 from Prowlarr or Hydra could also be seen there.

**Comment by Qstick** [comment]
> I'm thinking if maybe `indexerResponse.HttpResponse.HasHttpError` would be a better condition in case someone likes to use >= 200 <300 codes.

Yea, agree

File: src/NzbDrone.Core/Indexers/EzrssTorrentRssParser.cs [hunk 1]
```diff
@@ -20,6 +18,11 @@ public EzrssTorrentRssParser()
 
         protected override bool PreProcess(IndexerResponse indexerResponse)
         {
+            if (indexerResponse.HttpResponse.HasHttpError)
```
Comment (markus101) [inline src/NzbDrone.Core/Indexers/EzrssTorrentRssParser.cs:21]: I think we always want to call this, same as we did before to catch HTML content responses.

**Review by markus101 — CHANGES_REQUESTED** [review]
> Any info on what we are attempting to solve here?

This is why the template asks for this information, please don't nuke it.

File: src/NzbDrone.Core/Indexers/EzrssTorrentRssParser.cs (hunk 1 above)
Comment (markus101) [reply src/NzbDrone.Core/Indexers/EzrssTorrentRssParser.cs:21]: I understand, but EZRSS parser always checked this, unless there is a report of an issue then it should continue to do so.

Comment (markus101) [reply src/NzbDrone.Core/Indexers/EzrssTorrentRssParser.cs:21]: You don't want to revert it, just don't require an HTTP error to check it here.

Comment (markus101) [reply src/NzbDrone.Core/Indexers/EzrssTorrentRssParser.cs]: Only for EZRSS, it was the only one previously using `base.PreProcess`

**Review by markus101 — APPROVED** [review]

**Pull request merged**